numba/cuda/tests/cudapy/test_ufuncs.py

```diff
@@ -260,6 +260,18 @@ def test_bitwise_not_ufunc(self):
     # when the second argument is a negative. See the comment in
     # numba/tests/test_ufuncs.py for more details.
 
+    ############################################################################
+    # Mathematical Functions
+
+    def test_log_ufunc(self):
+        self.basic_ufunc_test(np.log, kinds='cf')
+
+    def test_log2_ufunc(self):
+        self.basic_ufunc_test(np.log2, kinds='cf')
+
+    def test_log10_ufunc(self):
+        self.basic_ufunc_test(np.log10, kinds='cf')
```

Comment:
I don't think these test functions will check values in the complex domain.

Reply:
I print the data that was generated in `basic_ufunc_test` and could see a few complex arrays there.

Reply:
The ufunc tests in `numba.tests.test_ufuncs` don't check values in the complex domain, but the CUDA ones have extra inputs added:

https://github.com/numba/numba/blob/03f2722e624d46223e3c95bc3910905b72d9a24d/numba/cuda/tests/cudapy/test_ufuncs.py#L33-L48